core/src/main/clojure/xtdb/expression/map.clj

```diff
@@ -126,15 +124,18 @@
             (.copyRow 0)))
 
         (let [build-key-cols (mapv #(vw/vec-wtr->rdr (.vectorFor rel-writer (str %))) build-key-col-names)]
-          (RelationMap. (update-keys build-fields str)
+          (RelationMap. allocator
+                        (update-keys build-fields str)
                         (map str build-key-col-names)
                         (update-keys probe-fields str)
                         (map str probe-key-col-names)
                         (boolean store-full-build-rel?)
                         rel-writer
                         build-key-cols
-                        hash->bitmap
                         (boolean nil-keys-equal?)
                         theta-expr
                         (update-keys param-types str)
-                        args))))))
+                        args
+                        (boolean with-nil-row?)
+                        96
```

Comment:
64/4 seemed fastest? also powers of 2, so may well be a situation where those are easier to work with